### PR TITLE
CONTRIBUTING.md: Add section about automatic testing and draft PRs

### PR DESCRIPTION
**Pull request by dhiller**
**What this PR does / why we need it**:

Add section about automated testing and draft pull requests to the CONTRIBUTING.md doc.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**This PR is ready for review, it's marked as draft, since the change is entirely markdown and CI signal is not needed.!**

/cc @fabiand @brianmcarey @xpivarc 

**Release note**:

```release-note
Add some context for initial contributors about automated testing and draft pull requests.
```

### CODE REVIEW

**Comment by dhiller** [comment]
@aburdenthehand would really appreciate you proofreading this if you have time. Thank you!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -60,6 +60,21 @@ run local without the need of CI. Have a look at the
...
+
+For more information about our CI please have a look at the [docs](https://github.com/kubevirt/project-infra/tree/main/docs) in project-infra repository.
+
+#### Consider opening your pull request as draft
+While many pull requests might be ready for review at the time they are created, some might not. This may be since the author might want to initiate a conversation or not be entirely sure whether the changes go into the right direction, or even because the changes are not complete.
+
+Please consider creating such PRs as [Draft Pull Requests](https://github.blog/2019-02-14-introducing-draft-pull-requests/). Draft PRs are skipped by CI, which will save CI resources. Also reviewers will not be automatically assigned to these, and everyone will understand that this pull request is not yet ready for review, which will save the waste of time looking at an unfinished pull request and wondering about it.
+
+After you mark your draft pull request ready for review, reviewers will get assigned and tests will be triggered if the ok-to-test label is present (see [above](CONTRIBUTING.md#automated-testing-of-pull-requests)).
```
Comment (aburdenthehand) [inline CONTRIBUTING.md:74]:
```suggestion
After you mark your draft pull request as ready for review, reviewers will be assigned and tests will be triggered if the ok-to-test label is present (see [above](CONTRIBUTING.md#automated-testing-of-pull-requests)).
```

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -60,6 +60,21 @@ run local without the need of CI. Have a look at the
 [Testing](docs/getting-started.md#testing)
 section in the [Developer Guide](docs/getting-started.md).
 
+#### Automated testing of pull requests
+
+Automated testing is automatically triggered on pull requests (PRs) opened by members of the KubeVirt organization, with the exception of _[draft pull requests](CONTRIBUTING.md#consider-opening-your-pull-request-as-draft)_. Pull requests opened by new contributors are initially marked with the label [`needs-ok-to-test`](https://github.com/kubevirt/kubevirt/labels/needs-ok-to-test) and are not automatically tested. Test lanes will be created after a member of the KubeVirt organization adds [`/ok-to-test`](https://prow.ci.kubevirt.io/command-help#ok_to_test) on the PR.
```
Comment (brianmcarey) [inline CONTRIBUTING.md:65]:
```suggestion
Automated testing is triggered on pull requests (PRs) opened by members of the KubeVirt organization, with the exception of _[draft pull requests](CONTRIBUTING.md#consider-opening-your-pull-request-as-draft)_. Pull requests opened by new contributors are initially marked with the label [`needs-ok-to-test`](https://github.com/kubevirt/kubevirt/labels/needs-ok-to-test) and are not automatically tested. Test lanes will be created after a member of the KubeVirt organization adds [`/ok-to-test`](https://prow.ci.kubevirt.io/command-help#ok_to_test) on the PR.
```

Comment (brianmcarey) [reply CONTRIBUTING.md:65]: automatically here might be redundant

**Review by brianmcarey — COMMENTED** [review]
/lgtm

One tiny nit comment - but feel free to continue as is.

**Comment by dhiller** [comment]
> /lgtm
> 
> One tiny nit comment - but feel free to continue as is.

Thanks @brianmcarey! Updated, PTAL.

**Review by brianmcarey — COMMENTED** [review]
/lgtm

Thanks @dhiller

**Comment by dhiller** [comment]
/override pull-kubevirt-metrics-lint

Unrelated.

**Review by xpivarc — COMMENTED** [review]
/approve

**Pull request merged**